### PR TITLE
[Tiny PR] Add missing flags to `astro deploy`

### LINKED ISSUE #2767: Include name flag in deploy cli command
### Issue Type

Missing content for an existing feature

### Links to Affected Docs

When running `astro deploy -h` the `-n` flag appears

```
  -n, --deployment-name string   Name of the deployment to deploy to
```

But this is not listed in the docs currently:

https://docs.astronomer.io/astro/cli/astro-deploy#options

### The Issue or Context

Missing flag from `astro deploy` options section

### Fix or Proposed Change

Add a `-n` and the description to the docs page

### Additional Notes

_No response_

### Required Reviewers

_No response_

### PR DESCRIPTION
**Pull request by manmeetkaur**
Resolves #2767

### CODE REVIEW

File: astro/cli/astro-deploy.md [hunk 1]
```diff
@@ -51,10 +50,10 @@ astro deploy <options>
...
+| `<deployment-id>`         | Specifies the Deployment to deploy to and bypasses the Deployment selection prompt                                                                    | Any valid Deployment ID                                          |
+| `-d`, `--dags`            | Deploy only your `dags` directory. See [DAG-only deploys](deploy-code.md#dag-only-deploys)                                                            | None                                                             |
+| `-n`, `--deployment-name` | The name of the Deployment to deploy to. Use as an alternative to `<deployment-id>`.                                                                  | Any valid Deployment name.                                       |
+| `--description`           | A description for your code deploy. Descriptions appear in the Cloud UI in your Deployment's **Deploy History**.                                      | None                                                             |
```
Comment (lzdanski) [inline astro/cli/astro-deploy.md:56]:
```suggestion
| `-n`, `--deployment-name` | The name of the Deployment to deploy to. Use as an alternative to `<deployment-id>`                                                                   | Any valid Deployment name                                        |
| `--description`           | A description for your code deploy. Descriptions appear in the Cloud UI in your Deployment's **Deploy History**                                       | None                                                             |
```

Comment (lzdanski) [reply astro/cli/astro-deploy.md:56]: We've got somewhat inconsistent `.` usage in this table from before this PR. Very small detail.

File: astro/cli/astro-deploy.md [hunk 2]
```diff
@@ -63,41 +62,65 @@ astro deploy <options>
...
 
-# Deploy directly to a specific Deployment
-$ astro deploy ckvvfp9tf509941drl4vela81n
+```bash
+astro deploy
+```
 
-# The CLI automatically selects this Deployment for your Astro project
-$ astro deploy ckvvfp9tf509941drl4vela81n --save
+This command will prompt you to enter the Deployment number to which you wish to deploy.
```
Comment (lzdanski) [inline astro/cli/astro-deploy.md:75]:
```suggestion
This command prompts you to enter the Deployment ID where you wish to deploy. 
```

**Review by lzdanski — COMMENTED** [review]
Tiny edits!

File: astro/cli/astro-deploy.md (hunk 2 above)
Comment (manmeetkaur) [reply astro/cli/astro-deploy.md:75]: @lzdanski i am not sure what is the right term to use here, as we have to enter a serial number and not deployment id as a response to the prompt. I can show you today when we chat.

File: astro/cli/astro-deploy.md [hunk 3]
```diff
@@ -63,41 +62,65 @@ astro deploy <options>
 | `-t`,`--test`             | The filepath to an alternative pytest file or directory                                                                                               | Valid filepath within your Astro project                         |
 | `--workspace-id <string>` | In the prompt to select a Deployment, only show Deployments within this Workspace                                                                     | Any valid Workspace ID                                           |
 | `-i`, `--image-name`      | The name of a pre-built custom Docker image to use with your project. The image must be available from a Docker registry hosted on your local machine | A valid name for a pre-built Docker image based on Astro Runtime |
+| `-w`, `--wait`            | Wait for the Deployment to become healthy before ending the command                                                                                   | None                                                             |
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:65]:
```suggestion
| `-w`, `--wait`            | Wait for the Deployment to become healthy before completing the command.                                                                                   | None                                                             |
```

File: astro/cli/astro-deploy.md [hunk 4]
```diff
@@ -63,41 +62,65 @@ astro deploy <options>
...
+astro deploy --image-name your-custom-runtime-image
+```
+
+To deploy only DAGs from your Astro project to a specific Deployment:
+
+```bash
+astro deploy ckvvfp9tf509941drl4vela81n --dags
+```
+
+See [Deploy a custom Docker image](release-notes.md#deploy-a-custom-docker-image-with-new---image-name-flag) for more details.
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:107]:
```suggestion
```

File: astro/cli/astro-deploy.md [hunk 5]
```diff
@@ -63,41 +62,65 @@ astro deploy <options>
...
+astro deploy ckvvfp9tf509941drl4vela81n
+```
+
+To configure Astro CLI to automatically use a given Deployment and current working directory combination for future deploys:
+
+```bash
+astro deploy ckvvfp9tf509941drl4vela81n --save
 ```
 
+To use a Docker image from your local Docker registry to build your Astro project:
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:95]:
```suggestion
To [use a custom Docker image](release-notes.md#deploy-a-custom-docker-image-with-new---image-name-flag) from your local Docker registry to build your Astro project:
```

File: astro/cli/astro-deploy.md [hunk 6]
```diff
@@ -63,41 +62,65 @@ astro deploy <options>
...
 
-# Deploy only DAGs to a specific Deployment
-$ astro deploy ckvvfp9tf509941drl4vela81n --dags
+To deploy directly to a specific Deployment:
+
+```bash
+astro deploy ckvvfp9tf509941drl4vela81n
+```
+
+To configure Astro CLI to automatically use a given Deployment and current working directory combination for future deploys:
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:89]:
```suggestion
To configure the Astro CLI to use a given Deployment and directory as a default for future deploys:
```

File: astro/cli/astro-deploy.md [hunk 7]
```diff
@@ -63,41 +62,65 @@ astro deploy <options>
...
 | `--workspace-id <string>` | In the prompt to select a Deployment, only show Deployments within this Workspace                                                                     | Any valid Workspace ID                                           |
 | `-i`, `--image-name`      | The name of a pre-built custom Docker image to use with your project. The image must be available from a Docker registry hosted on your local machine | A valid name for a pre-built Docker image based on Astro Runtime |
+| `-w`, `--wait`            | Wait for the Deployment to become healthy before ending the command                                                                                   | None                                                             |
 
 ## Examples
 
-```sh
-# List of Deployments appears
-$ astro deploy
+To view a list of available Deployments in your current Workspace, run the following command:
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:69]: We can delete this first example since it's identical to the "Usage"

File: astro/cli/astro-deploy.md [hunk 8]
```diff
@@ -63,41 +62,65 @@ astro deploy <options>
...
 
-# The CLI automatically selects this Deployment for your Astro project
-$ astro deploy ckvvfp9tf509941drl4vela81n --save
+This command will prompt you to enter the Deployment number to which you wish to deploy. 
 
-# The CLI looks for a Docker image with a matching name in your local Docker registry and builds your project with it
-$ astro deploy --image-name my-custom-runtime-image
+:::info
+
+If you have uncommitted changes in your working directory, you must use `-f` or `--force` flag with this command, or commit changes to your code repository.
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:79]:
```suggestion
If you have uncommitted changes in your working directory, you must use the `-f` or `--force` flag with this command or commit changes to your code repository.
```

**Review by jwitz — APPROVED** [review]

**Pull request merged**